### DISCUSSION TITLE
Train Dataloader

### DISCUSSION THREAD

**Discussion by alexliap** [SalesforceAIResearch/uni2ts#126] [Q&A]
Could someone explain to me how the train dataloader works?

I see the arguments batch size and batch size factor. what is the synergy here?

For example, lets say i have an hourly data set of one year (24h*365d).

What values should i put to batch size, batch size factor and num_of_barches_per_epoch in order to pass the entire dataset to the model in one epoch?

**Comment by chenghaoliu89** [comment]
Hi @alexliap, uni2ts have two dataloder. 1. the original pytorch dataloader with batch size = batch size * factor,  each sample has varying length, 2. moirai dataloader with batch size = batch size, which transform the original batch data from pytorch dataloader to packed batch samples. and them each row of batch data has equal length.
